src/System.Drawing.Common/src/System/Drawing/Image.Windows.cs

```diff
@@ -134,6 +134,8 @@ public void Save(string filename, ImageCodecInfo encoder, EncoderParameters enco
                 throw new ArgumentNullException(nameof(filename));
             if (encoder == null)
                 throw new ArgumentNullException(nameof(encoder));
+            if (!Directory.Exists(Path.GetDirectoryName(filename)))
+                throw new DirectoryNotFoundException(nameof(filename));
```

Comment:
Note that wierdly, the DirectoryNotFoundException constructor takes a message, not a path.

See for example this code elsewhere
```c#
            if (!new DirectoryInfo(Path.GetDirectoryName(FullName)).Exists)
                throw new DirectoryNotFoundException(SR.Format(SR.IO_PathNotFound_Path, FullName));
```
Which incidentally results in something like "Could not find a part of the path 'c:\foo\bar\myfile.bmp'." I think it would be nicer if it was "Could not find a part of the path 'c:\foo\bar'.", ie., instead of `FullName` it ideally would pass in the result of `GetDirectoryName(FullName)`. We could probably fix that.

Reply:
It would not be unreasonable to add a Debug.Assert into the DirectoryNotFoundException constructor to catch cases like this in our own code (for example `Debug.Assert(message.Contains(' '));`

Reply:
One more comment - please add a test...

Reply:
Ok, didn't know the error messages are translated. Will submit pull request tomorrow to dotnet/runtime.